### PR TITLE
Fix ARNs parsing confusion when the account-ID field contained a slash or the resource-ID a colon.

### PR DESCRIPTION
**Pull request by arielshaqed**

### CODE REVIEW

File: pkg/auth/arn_test.go [hunk 1]
```diff
@@ -38,12 +32,18 @@ func TestParseARN(t *testing.T) {
 			Region:     "",
 			AccountID:  "b",
 			ResourceID: "myrepo"}},
-		{Input: "arn:lakefs:repos::/myrepo", Arn: auth.Arn{
+		{Input: "arn:lakefs:repos:::myrepo", Arn: auth.Arn{
 			Partition:  "lakefs",
 			Service:    "repos",
 			Region:     "",
 			AccountID:  "",
 			ResourceID: "myrepo"}},
```
Comment (itaiad200) [inline pkg/auth/arn_test.go:40]: Shouldn't this be:
```suggestion
			ResourceID: ":myrepo"}},
```

**Review by ozkatz — APPROVED** [review]
Thanks, LGTM!

File: pkg/auth/arn_test.go [hunk 2]
```diff
@@ -82,8 +82,7 @@ func TestArnMatch(t *testing.T) {
 		{"arn:lakefs:repos::b:myrepo", "arn:lakefs:repos::b:myrepo", true},
 		{"arn:lakefs:repos::b:*", "arn:lakefs:repos::b:myrepo", true},
 		{"arn:lakefs:repos::b:myrepo", "arn:lakefs:repos::b:*", false},
-		{"arn:lakefs:repos::b/myrepo", "arn:lakefs:repos::b/*", false},
-		{"arn:lakefs:repos::b/*", "arn:lakefs:repos::b/myrepo", true},
```
Comment (itaiad200) [inline pkg/auth/arn_test.go:86]: Can you fix these examples instead of deleting them?

**Review by itaiad200 — APPROVED** [review]
LGTM, 2 minor comments

File: pkg/auth/arn_test.go (hunk 2 above)
Comment (arielshaqed) [reply pkg/auth/arn_test.go:86]: Sorry, but again no: these test the previous behaviour of allowing a `/` to separate account-ID from resource-ID.  This is simply no longer allowed.

**Review by arielshaqed — COMMENTED** [review]
Thanks for a very speedy turnaround!

I shall rephrase the CHANGELOG entry to allow users who might rely on the existing behaviour to understand that it is now gone -- and pull.

**Comment by arielshaqed** [comment]
> Thanks for a very speedy turnaround!
> 
> I shall rephrase the CHANGELOG entry to allow users who might rely on the existing behaviour to understand that it is now gone -- and pull.

Actually it's about as clear as can be.  And I doubt anyone is using it, given that there are currently no useful account IDs in lakeFS ARNs.

**Pull request merged**